homeassistant/components/nightscout/__init__.py

```diff
@@ -0,0 +1,66 @@
+"""The Nightscout integration."""
+import asyncio
+import logging
+
+from py_nightscout import Api as NightscoutAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import SLOW_UPDATE_WARNING
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORMS = ["sensor"]
+_API_TIMEOUT = SLOW_UPDATE_WARNING - 1
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Nightscout component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Nightscout from a config entry."""
+    host = entry.data[CONF_HOST]
+
+    api = NightscoutAPI(host)
+    status = await api.get_server_status()
+
+    hass.data[DOMAIN][entry.entry_id] = api
+
+    device_registry = await dr.async_get_registry(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, status.name)},
+        manufacturer="NighScout",
+        name=status.name,
+        sw_version=status.version,
+    )
```

Comment:
wonder if this should have entry_type= "service"?

Reply:
Done! Also changed the identifier to use the `host` URL and fixed the `manufacturer`